Web: Missing Refresh Token

### Checklist

- [X] The issue can be reproduced in the [auth0_flutter sample app](https://github.com/auth0-samples/auth0-flutter-samples/tree/main/sample) (or N/A).
- [X] I have looked into the [Readme](https://github.com/auth0/auth0-flutter/tree/main/auth0_flutter#readme), [Examples](https://github.com/auth0/auth0-flutter/blob/main/auth0_flutter/EXAMPLES.md), and [FAQ](https://github.com/auth0/auth0-flutter/blob/main/auth0_flutter/FAQ.md) and have not found a suitable solution or answer.
- [X] I have looked into the [API documentation](https://pub.dev/documentation/auth0_flutter/latest/) and have not found a suitable solution or answer.
- [X] I have searched the [issues](https://github.com/auth0/auth0-flutter/issues) and have not found a suitable solution or answer.
- [X] I have searched the [Auth0 Community](https://community.auth0.com) forums and have not found a suitable solution or answer.
- [X] I agree to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md).

### Description

I found a problem using refreshtokens and localstorage on web implementation. I not 100% sure if im doing it wrong or if this is a bug, because i cannot give an audience for onLoad function.


### Reproduction

This is the inital conf:

```
      final credentials = await auth0Web.onLoad(
        cacheLocation: CacheLocation.localStorage,
        useRefreshTokens: true,
      );
```
and here the Login happens:
```
final credentials = await auth0Web.loginWithPopup(
        audience: Config.auth0Audience,
      );
```
The data is saved in LocalStorage as expected with audience information.

When I rerun the application i get the following WebException from onLoad():

[ERROR] | 11:26:38 466ms | Missing Refresh Token (audience: '', scope: 'openid profile email offline_access')

So there is an error that we do not have an audience here. Where can i config the audience in onLoad? Is this an misusing on my side?

### Additional context

_No response_

### auth0_flutter version

1.2.0

### Flutter version

3.10.2

### Platform

Web

### Platform version(s)

1.2.0

--- a/auth0_flutter/lib/auth0_flutter_web.dart
+++ b/auth0_flutter/lib/auth0_flutter_web.dart
@@ -32,6 +32,11 @@ class Auth0Web {
   /// learn more about these parameters.
   /// * See the [ClientOptions] type for the full description of the remaining
   /// parameters for this method.
+  /// * [audience] relates to the API Identifier you want to reference in your
+  /// access tokens. See [API settings](https://auth0.com/docs/get-started/apis/api-settings)
+  /// to learn more.
+  /// * [scopes] defaults to `openid profile email`. You can override these
+  /// scopes, but `openid` is always requested regardless of this setting.
   Future<Credentials?> onLoad(
       {final int? authorizeTimeoutInSeconds,
       final CacheLocation? cacheLocation,
@@ -44,7 +49,9 @@ class Auth0Web {
       final bool? useCookiesForTransactions,
       final bool? useFormData,
       final bool? useRefreshTokens,
-      final bool? useRefreshTokensFallback}) async {
+      final bool? useRefreshTokensFallback,
+      final String? audience,
+      final Set<String>? scopes}) async {
     await Auth0FlutterWebPlatform.instance.initialize(
         ClientOptions(
             account: _account,
@@ -59,7 +66,9 @@ class Auth0Web {
             useCookiesForTransactions: useCookiesForTransactions,
             useFormData: useFormData,
             useRefreshTokens: useRefreshTokens,
-            useRefreshTokensFallback: useRefreshTokensFallback),
+            useRefreshTokensFallback: useRefreshTokensFallback,
+            audience: audience,
+            scopes: scopes),
         _userAgent);
 
     if (await hasValidCredentials()) {

--- a/auth0_flutter/lib/src/web/extensions/client_options_extensions.dart
+++ b/auth0_flutter/lib/src/web/extensions/client_options_extensions.dart
@@ -1,5 +1,6 @@
 import 'package:auth0_flutter_platform_interface/auth0_flutter_platform_interface.dart';
 import '../js_interop.dart';
+import '../js_interop_utils.dart';
 
 extension ClientOptionsExtension on ClientOptions {
   Auth0ClientOptions toAuth0ClientOptions(final UserAgent userAgent) =>
@@ -19,5 +20,8 @@ extension ClientOptionsExtension on ClientOptions {
           useCookiesForTransactions: useCookiesForTransactions,
           useFormData: useFormData,
           useRefreshTokens: useRefreshTokens,
-          useRefreshTokensFallback: useRefreshTokensFallback);
+          useRefreshTokensFallback: useRefreshTokensFallback,
+          authorizationParams: JsInteropUtils.stripNulls(AuthorizationParams(
+              audience: audience,
+              scope: scopes?.isNotEmpty == true ? scopes?.join(' ') : null)));
 }

--- a/auth0_flutter/lib/src/web/js_interop.dart
+++ b/auth0_flutter/lib/src/web/js_interop.dart
@@ -79,6 +79,7 @@ class Auth0ClientOptions {
       {required final Auth0ClientInfo clientInfo,
       required final String domain,
       required final String clientId,
+      final String? audience,
       final int? authorizeTimeoutInSeconds,
       final String? cacheLocation,
       final String? cookieDomain,
@@ -90,7 +91,8 @@ class Auth0ClientOptions {
       final bool? useCookiesForTransactions,
       final bool? useFormData,
       final bool? useRefreshTokens,
-      final bool? useRefreshTokensFallback});
+      final bool? useRefreshTokensFallback,
+      final AuthorizationParams? authorizationParams});
 }
 
 @JS()

--- a/auth0_flutter/lib/src/web/js_interop.dart
+++ b/auth0_flutter/lib/src/web/js_interop.dart
@@ -79,7 +79,6 @@ class Auth0ClientOptions {
       {required final Auth0ClientInfo clientInfo,
       required final String domain,
       required final String clientId,
-      final String? audience,
       final int? authorizeTimeoutInSeconds,
       final String? cacheLocation,
       final String? cookieDomain,

--- a/auth0_flutter_platform_interface/lib/src/web/client_options.dart
+++ b/auth0_flutter_platform_interface/lib/src/web/client_options.dart
@@ -95,6 +95,16 @@ class ClientOptions {
   /// The configuration for validating ID tokens.
   final IdTokenValidationConfig? idTokenValidationConfig;
 
+  /// The default audience to be used for requesting API access.
+  final String? audience;
+
+  /// The default scopes to be used on authentication requests.
+  ///
+  /// This defaults to `openid profile email` if not specified.
+  ///
+  /// Note: The openid scope is always applied regardless of this setting.
+  final Set<String>? scopes;
+
   ClientOptions(
       {required this.account,
       this.authorizeTimeoutInSeconds,
@@ -107,5 +117,7 @@ class ClientOptions {
       this.useFormData,
       this.useRefreshTokens,
       this.useRefreshTokensFallback,
-      this.idTokenValidationConfig});
+      this.idTokenValidationConfig,
+      this.audience,
+      this.scopes});
 }

